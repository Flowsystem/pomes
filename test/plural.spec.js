import React from 'react';
import { mount } from 'enzyme';
import Plural from 'plural';
import I18nProvider from 'component/component';
import toJson from 'enzyme-to-json';

describe('Plural', () => {
  it('render singular message', () => {
    const translations = {};
    const message = mount(
      <I18nProvider translations={translations} lang="en" initialLang="en" initialized>
        <Plural
          texts={["You have {count} message", "You have {count} messages"]}
          condition="count"
          values={{ count: 1 }}
          comment="Foo"
        />
      </I18nProvider>,
    );

    expect(toJson(message)).toMatchSnapshot();
  });

  it('render plural message', () => {
    const translations = {};
    const message = mount(
      <I18nProvider translations={translations} lang="en" initialLang="en" initialized>
        <Plural
          texts={["You have {count} message", "You have {count} messages"]}
          condition="count"
          values={{ count: 2 }}
          comment="Foo"
        />
      </I18nProvider>,
    );

    expect(toJson(message)).toMatchSnapshot();
  });
});
