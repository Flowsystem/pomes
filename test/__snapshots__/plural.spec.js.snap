// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plural render plural message 1`] = `
<I18nProvider
  dispatch={null}
  fallbackLang={null}
  initialLang="en"
  initialized={true}
  lang="en"
  translations={Object {}}
  useReducer={false}
>
  <Plural
    comment="Foo"
    condition="count"
    texts={
      Array [
        "You have {count} message",
        "You have {count} messages",
      ]
    }
    values={
      Object {
        "count": 2,
      }
    }
  >
    You have 2 messages
  </Plural>
</I18nProvider>
`;

exports[`Plural render singular message 1`] = `
<I18nProvider
  dispatch={null}
  fallbackLang={null}
  initialLang="en"
  initialized={true}
  lang="en"
  translations={Object {}}
  useReducer={false}
>
  <Plural
    comment="Foo"
    condition="count"
    texts={
      Array [
        "You have {count} message",
        "You have {count} messages",
      ]
    }
    values={
      Object {
        "count": 1,
      }
    }
  >
    You have 1 message
  </Plural>
</I18nProvider>
`;
